### PR TITLE
Fix the massive discontinuities I saw in terrain generation.

### PR DESCRIPTION
**Pull request by kieve**
Fixes those giant cliffs we saw... I think.

### CODE REVIEW

**Comment by crazy2be** [comment]
How does this work?

**Comment by crazy2be** [comment]
Like why does this fix it? It seems like magic.

**Comment by kieve** [comment]
See http://www.java-gaming.org/topics/2d-simplex-noise-infinite-terrain-generation/29738/msg/273519/view.html#msg273519

The gist of it is that the noise is calculated form "simplex" locations. triangle positions. To get smooth noise, you need to sample these triangles next to each other. But using integers we're skipping triangles in some cases, and thus skipping the smooth transition between them.

**Comment by kieve** [comment]
Turns out this doesn't fix it. Need to do some more research.

**Comment by kieve** [comment]
Should be fixed now.

File: server/lib/mapgen/maps/simplexHills.go [hunk 1]
```diff
@@ -12,12 +12,12 @@ type SimplexHills struct {
...
 func NewSimplexHills(seed int64) *SimplexHills {
 	sh := new(SimplexHills)
-	sh.simplexNoise = noise.NewSimplex(10, 0.4, seed)
+	sh.simplexNoise = noise.NewSimplex(4, 0.8, seed)
 	return sh
 }
 
 func (sh *SimplexHills) heightAt(x, z float64) float64 {
-	height := sh.simplexNoise.Noise2(x, z)
+	height := sh.simplexNoise.Noise2(x/128, z/128)
```
Comment (crazy2be) [inline server/lib/mapgen/maps/simplexHills.go]: These changes still needed?

File: server/lib/game/chunkGenerator.go [hunk 2]
```diff
@@ -166,7 +166,7 @@ func EachChunkNearby(wc coords.World, cb func(cc coords.Chunk, priority int)) {
 	}
 
 	cc := wc.Chunk()
-	eachWithin(cc, 2, 1, 2, func(newCC coords.Chunk, dist int) {
+	eachWithin(cc, 10, 3, 10, func(newCC coords.Chunk, dist int) {
```
Comment (crazy2be) [inline server/lib/game/chunkGenerator.go]: Woah, that view distance.

How much bandwidth does this use?

Comment (kieve) [inline server/lib/game/chunkGenerator.go]: I'm reverting this now. This was changed just so I could quickly check far away for issues.

File: server/lib/mapgen/maps/simplexHills.go (hunk 1 above)
Comment (kieve) [inline server/lib/mapgen/maps/simplexHills.go]: Ideally, yeah. Besides, using less octaves but reducing the input by a factor increases the performance by a lot.

**Pull request merged**